Sources/Network/URLSessionNetworkStack.swift

```diff
@@ -136,7 +136,7 @@ public extension Network {
 
                 guard let httpResponse = response as? HTTPURLResponse else {
                     // don't retry this error since this should "never" happen
-                    completion(.failure(.badResponse))
+                    completion(.failure(Network.Error(type: .badResponse, response: nil)))
```

Comment:
we should propagate the `response` here, as it might be non nil

And if we nest it on the `Error` enum, this would then be `completion(.failure(.badResponse(response: response)))` (or even simply `completion(.failure(.badResponse(response)))`)